src/engine/controls/cuecontrol.cpp

```diff
@@ -136,6 +137,73 @@ CueControl::~CueControl() {
     qDeleteAll(m_hotcueControls);
 }
 
+mixxx::audio::FramePos CueControl::nextTrigger(bool reverse,
+        mixxx::audio::FramePos currentPosition,
+        mixxx::audio::FramePos* pTargetPosition,
+        mixxx::audio::FrameDiff_t requested_frame) {
+    VERIFY_OR_DEBUG_ASSERT(pTargetPosition) {
+        return mixxx::audio::kInvalidFramePos;
+    }
+    *pTargetPosition = mixxx::audio::kInvalidFramePos;
+    mixxx::audio::FramePos triggerPosition = mixxx::audio::kInvalidFramePos;
+    HotcueControl* pNextJump = nullptr;
+    for (const auto& pControl : std::as_const(m_hotcueControls)) {
+        if (!pControl->getCue() ||
+                pControl->getStatus() != HotcueControl::Status::Active ||
+                pControl->getCue()->getType() != mixxx::CueType::Jump ||
+                !pControl->getEndPosition().isValid()) {
+            continue;
+        }
+
+        if (!reverse
+                // Saved jumps store the position to jump from as their end
+                // position
+                && pControl->getEndPosition() > currentPosition &&
+                (pControl->getEndPosition() < triggerPosition ||
```

Comment:
I am not sure what the result of comapring with `mixxx::audio::kInvalidFramePos (nan)` is in our fast-math code. 
It is better to avoid it and swap the checks.